### PR TITLE
add invoice tdd and methods

### PR DESCRIPTION
**Pull request by IcelandicGambit**

### CODE REVIEW

File: test/sales_analyst_test.rb [hunk 1]
```diff
@@ -158,4 +158,34 @@ def test_percentage_of_unshipped_invoices
...
+                              :items     => "./data/items.csv",
+                              :merchants => "./data/merchants.csv",
+                              :invoices => "./data/invoices.csv",
+                              :transactions => "./data/transactions.csv",
+                              :invoice_items => "./data/invoice_items.csv",
+                              :customers => "./data/customers.csv"
+                              })
+    sales_analyst = sales_engine.analyst
+    expected = sales_analyst.invoice_total(1)
+    assert_equal 348956.00, expected
```
Comment (krislitman) [inline test/sales_analyst_test.rb:189]: We may just need to double check that this number will pass with the spec harness as well!

**Comment by krislitman** [comment]
How I feel about all the naming conventions in this project..... officespace

File: test/sales_analyst_test.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (klatour324) [reply test/sales_analyst_test.rb:189]: This is actually a good suggestion from the hound to make it easier to read the magnitude of the number. Let's refactor this and make sure in our refactoring we follow through everywhere large integer or float values exist with this styling.

**Comment by klatour324** [comment]
We could definitely do better with naming things. Great work getting through this. Let's check That our number for sales analyst invoice total works.

**Comment by IcelandicGambit** [comment]
Naming conventions and refactoring left, one iteration 1 spec harness test to pass.

**Pull request merged**